src/main/python/ayab/plugins/ayab_plugin/ayab_control.py

```diff
@@ -370,82 +376,95 @@ def __middlecoltwice(self, lineNumber, imgHeight, lenImgExpanded):
 
         # Double the line minus the 2 you save on the begin
         # and end of each imgRow
-        passesPerRow = 2 * self.__numColors - 2
+        passesPerRow = 2 * self._numColors - 2
+
+        q, r = divmod(lineNumber, passesPerRow)
+        firstCol = (r == 0)
+        lastCol = (r == passesPerRow - 1)
 
-        imgRow = self.__startLine + lineNumber // passesPerRow
+        imgRow = self._startLine + q
 
-        if self.__infRepeat:
+        if self._infRepeat:
             imgRow = imgRow % imgHeight
 
-        if imgRow % 2 != 0:
-            color = ((lineNumber % passesPerRow) + 1) // 2
+        if firstCol or lastCol:
+            color = (lastCol + q) % 2
```

Comment:
This changes corrects what I believe to be a bug that would occur with infinite repeats and image files with an odd number of rows.